libs/barista-components/sunburst/src/sunburst.ts

```diff
@@ -0,0 +1,261 @@
+      this.filledSeries,
+      getSelectedId(this.filledSeries, this._selected),
+    );
+    this.slices = getSlices(nodesWithState);
+
+    if (this._selected && this._selected.length) {
+      this.selectedLabel = this._selected.slice(-1)[0].label ?? '';
+      this.selectedValue = this._selected.slice(-1)[0].value ?? 0;
+      this.selectedRelativeValue =
+        this._selected.slice(-1)[0].valueRelative ?? 0;
+    } else {
+      this.selectedLabel = this.noSelectionLabel;
+      this.selectedValue = getValue(this.filledSeries);
+      this.selectedRelativeValue = 1;
+    }
+  }
+  /** Programmatically close the overlay. */
+  closeOverlay(): void {
+    this._dismissOverlay();
+  }
+
+  /** Puts the overlay in place. */
+  openOverlay(node: DtSunburstSlice): void {
+    const origin = this.getOriginFromSlice(node);
+
+    this._dismissOverlay();
+    this._createOverlay(origin, node.data);
+  }
+
+  /** Creates the overlay and attaches it. */
+  private _createOverlay(
+    origin: { x: number; y: number },
+    node: DtSunburstNodeInternal,
+  ): void {
+    // If we do not have an overlay defined, we do not need to attach it
+    if (!this._overlay) {
+      return;
+    }
+
+    // Create the template portal
+    if (!this._portal) {
+      // tslint:disable-next-line: no-any
+      this._portal = new TemplatePortal<any>(
+        this._overlay,
+        this._viewContainerRef,
+        { $implicit: node },
+      );
+    }
+
+    const positionStrategy = this._overlayService
+      .position()
+      .flexibleConnectedTo(origin)
+      .setOrigin(origin)
+      .withPositions([
+        {
+          originX: 'center',
+          originY: 'center',
+          overlayX: 'center',
+          overlayY: 'center',
+        },
+      ])
+      .withFlexibleDimensions(true)
+      .withPush(false)
+      .withGrowAfterOpen(true)
+      .withViewportMargin(0)
+      .withLockedPosition(false);
+
+    const overlayConfig = new OverlayConfig({
+      positionStrategy,
+      panelClass: OVERLAY_PANEL_CLASS,
+      backdropClass: 'cdk-overlay-transparent-backdrop',
+    });
+    this._overlayRef = this._overlayService.create(overlayConfig);
+
+    // If the portal is not yet attached to the overlay, attach it.
+    if (!this._overlayRef.hasAttached()) {
+      this._overlayRef.attach(this._portal);
+    }
+    dtSetUiTestAttribute(
+      this._overlayRef.overlayElement,
+      this._overlayRef.overlayElement.id,
+      this._elementRef,
+      this._config,
+    );
+  }
+
+  /** Dismisses the overlay. */
+  private _dismissOverlay(): void {
+    if (this._overlayRef) {
+      this._overlayRef.detach();
+      this._portal = null;
+    }
+  }
+  /**
+   * Calculates the origin for the overlay based on the target of the
+   * event passed into it.
+   */
+  private getOriginFromSlice(slice: DtSunburstSlice): { x: number; y: number } {
+    const rect: DOMRect = this.svgEl.nativeElement.getBoundingClientRect();
```

Comment:
Before accessing the nativeElement, could you please insert a platform check. Otherwise this will fail in a universal rendering environment.